packages/salesforcedx-utils/test/jest/cli/cliCommandExecution.test.ts

```diff
@@ -0,0 +1,203 @@
+    const cliCommandExecution = new CliCommandExecution(
+      testCommand,
+      testChildProcess,
+      testCancelationToken
+    );
+    expect(cliCommandExecution).toBeInstanceOf(CliCommandExecution);
+    expect(subscribeSpy).toHaveBeenCalledTimes(3);
+    expect(intervalSpy).toHaveBeenCalledWith(CANCELATION_INTERVAL);
+  });
+
+  describe('Subscribe handlers.', () => {
+    it('Should call timer unsubscribe on exit.', () => {
+      const cliCommandExecution = new CliCommandExecution(
+        testCommand,
+        testChildProcess,
+        testCancelationToken
+      );
+      expect(cliCommandExecution).toBeDefined();
+      const exitSubscribeHandler = subscribeSpy.mock.calls[0][0];
+      exitSubscribeHandler();
+      expect(unsubscribeSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('Should call timer unsubscribe on error.', () => {
+      const cliCommandExecution = new CliCommandExecution(
+        testCommand,
+        testChildProcess,
+        testCancelationToken
+      );
+      expect(cliCommandExecution).toBeDefined();
+      const exitSubscribeHandler = subscribeSpy.mock.calls[1][0];
+      exitSubscribeHandler();
+      expect(unsubscribeSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Verify child process.', () => {
+    it('Should fail to create if child has no pid.', () => {
+      testChildProcess.pid = undefined;
+      expect(() => {
+        new CliCommandExecution(
+          testCommand,
+          testChildProcess,
+          testCancelationToken
+        );
+      }).toThrowError(NO_PID_ERROR);
+    });
+    it('Should fail to create if child has no stdout.', () => {
+      testChildProcess.stdout = undefined;
+      expect(() => {
+        new CliCommandExecution(
+          testCommand,
+          testChildProcess,
+          testCancelationToken
+        );
+      }).toThrowError(NO_STDOUT_ERROR);
+    });
+    it('Should fail to create if child has no stderr.', () => {
+      testChildProcess.stderr = undefined;
+      expect(() => {
+        new CliCommandExecution(
+          testCommand,
+          testChildProcess,
+          testCancelationToken
+        );
+      }).toThrowError(NO_STDERR_ERROR);
+    });
+  });
+
+  describe('kill on timeout.', () => {
+    let logSpy: jest.SpyInstance;
+
+    beforeEach(() => {
+      logSpy = jest.spyOn(console, 'log');
+    });
+    it('Should be able to successfully kill child process.', async () => {
+      testCancelationToken.isCancellationRequested = true;
+      const cliCommandExecution = new CliCommandExecution(
+        testCommand,
+        testChildProcess,
+        testCancelationToken
+      );
+      expect(cliCommandExecution).toBeInstanceOf(CliCommandExecution);
+      expect(subscribeSpy).toHaveBeenCalledTimes(3);
+      const timoutHandler = subscribeSpy.mock.calls[2][0];
+      const timeoutPromise = timoutHandler();
+      expect(treeKillMocked).toHaveBeenCalledTimes(1);
```

Comment:
note how we can use the treeKillMocked here to verify that an dependent module was called.